### ISSUE TITLE
egs_chamber crashes when reading in cross-section data

### ISSUE THREAD

**Issue by MartinMartinov** [nrc-cnrc/EGSnrc#260]
After pulling recent changes (February's commits) and recompiling my HEN_HOUSE and EGS_HOME areas, I'm finding my egs_chamber simulations crash when reading certain data files, namely spinms.data and the rayleigh data set (egslog doesn't name which file, but it happens when rayleigh scattering is turned on).  It happens with many of the input files that have worked over the last year for me.  I also tried running example1.egsinp on cavity and got the same error.  This is the pre-crash output:

    Reading spin data base from /home/martinov/HEN_HOUSE_github/EGSnrc/HEN_HOUSE/data/spinms.data
    EGSnrc spin data, version 2.0
    Data generated on a machine with 1234 endianess
    The endianess of this CPU is 1234
    Ranges:      1.00   100.00  0.30054  1.00000

      medium    1 ..................... Segmentation fault

I checked my previous log files from before the reinstall, and I find that the output should look like this:

    Reading spin data base from /home/martinov/HEN_HOUSE_github/EGSnrc/HEN_HOUSE/data/spinms.data
    EGSnrc spin data, version 2.0
    Data generated on a machine with 1234 endianess
    The endianess of this CPU is 1234
    Ranges:      1.00   100.00  0.30054  1.00000

      medium    1 .....................  done
      medium    2 .....................  done
      medium    3 .....................  done
      medium    4 .....................  done
      medium    5 .....................  done

Any ideas on what my issue is?

Config flags that might be important:

    F77 = gfortran
    F77_LINK = gfortran
    FCFLAGS = -fPIC
    FOPT = -O2
    FDEBUG = -g
    FLIBS =
    FOUT = -o
    FOBJE = o

    CC = gcc
    C_FLAGS = -O2 -fPIC
    COUT = -o

    CXX = g++
    opt = -O3 -ffast-math
    DEF1 =  -fPIC

Compiler version:

    gcc version 4.4.7 20120313 (Red Hat 4.4.7-4) (GCC)

**Comment by rtownson** [comment]
This was introduced by the rest mass initialization fix yesterday. PR #261 is now pending.

**Comment by ftessier** [comment]
This is because we omitted to update the common block definition in the C interface to initialize the rest mass as well. Working on it:

[egs_c_interface1.macros](https://github.com/nrc-cnrc/EGSnrc/blob/master/HEN_HOUSE/interface/egs_c_interface1.macros#L199-L205)
[egs_c_interface2.macros](https://github.com/nrc-cnrc/EGSnrc/blob/805881c9fb1a517b46e00acfd2f83b6cb5e92e52/HEN_HOUSE/interface/egs_c_interface2.macros#L229-L235)

**Comment by ftessier** [comment]
Checking with `grep -iR "replace.*useful/" HEN_HOUSE/*` would seem to indicate that there are no other independent instances of the COMIN/USEFUL macro in the repository:
```
HEN_HOUSE/interface/egs_c_interface1.macros:REPLACE {;COMIN/USEFUL/;} WITH {
HEN_HOUSE/interface/egs_c_interface2.macros:REPLACE {;COMIN/USEFUL/;} WITH {
HEN_HOUSE/src/egsnrc.macros:REPLACE {;COMIN/USEFUL/;} WITH
```

**Comment by ftessier** [comment]
This was fixed in pull request #261 and cheery-picked to master via #262.

:1st_place_medal: to @MartinMartinov for reporting the issue!

**Comment by MartinMartinov** [comment]
Well done!  Really good turn around time for this fix, I'll try it first thing in the morning.
